router/realm.go

```diff
@@ -183,14 +183,14 @@ func (r *realm) close() {
 
 	// Kick all clients off.  Sending shutdownGoodbye causes client message
 	// handlers to exit without sending meta events.
-	sync := make(chan struct{})
+	syncChan := make(chan struct{})
 	r.actionChan <- func() {
 		for _, c := range r.clients {
 			c.EndRecv(shutdownGoodbye)
 		}
-		close(sync)
+		close(syncChan)
 	}
-	<-sync
+	<-syncChan
```

Comment:
Same question about possibly providing a context to cancel the wait.

Reply:
It is a realm closing and here is waiting for all clients to close normally. 
I'm not so deeply familiar with all codebase and why that was done like it is...